README.md

```diff
@@ -27,3 +35,25 @@ docker_build('gcr.io/windmill-test-containers/integration/onewatch',
 ```
 
 This live update will cause the `go install` to be run in the container every time anything in the `.` path locally changes. After the `go install` is run, `./restart.sh` will be run. This will kill the original entrypoint, and restart it, effectively simulating the `container_restart()` functionality on Docker.
+
+## Non-Supported Cases
+
+### Containers Without Shell
+At the risk of stating the obvious: your container needs to have shell available to run these shell scripts.
+
+These scripts look for a shell at `/bin/sh`; if your shell lives elsewhere, you have to explicitly invoke the scripts with that shell, e.g.:
+```python
+docker_build(...,
+    entrypoint=['/busybox/sh', './start.sh', 'path/to/binary'],
+    ...
+)
+```
+
+### Arbitrary Shell Commands/Child Processes
+You may encounter difficulties using these scripts to run arbitrary shell commands. E.g. an invocation like:
+```bash
+./start.sh /bin/sh -c 'ls | path/to/binary'
+```
+will not be handled correctly. We recommend only using `start.sh` to run commands passable as program name + list of args.
```

Comment:
out of curiosity, why won't it be handled correctly?

Reply:
[Here's a test case](https://github.com/tilt-dev/abc123/tree/start.sh-shell-weirdness) (run `tilt up fe`). I've got this entrypoint:
```
entrypoint=['/start.sh', 'sh', '-c', 'echo hello from shell 👋; /go/bin/fe'],
```

In the pod, I see these processes:
```
UID        PID  PPID  C STIME TTY          TIME CMD
root         1     0  0 20:11 ?        00:00:00 /bin/sh /start.sh sh -c echo hello from shell ????;
root         7     1  0 20:11 ?        00:00:00 sh -c echo hello from shell ????; /go/bin/fe
root         8     7  0 20:11 ?        00:00:00 /go/bin/fe
```
`process.txt` (i.e. the process we're gonna kill) points to PID 7; but when I call `/restart.sh`, I get a "bind address already in use" error; afaict killing PID 7 doesn't kill it's child process PID 8 (i.e. the actual binary), and when I try to rerun the binary, it complains because its port is already taken. [Here's a snapshot](https://cloud.tilt.dev/snapshot/AdLDlO4LwnWuz6LfWXY=).

FWIW I have [a version of this script that I thiiiiiink solves this problem](https://github.com/tilt-dev/abc123/tree/start.sh-shell-working)? By which I mean, last time I worked on this, I abandoned this branch b/c it was still broken but I can't reproduce the brokenness now...?

Reply:
Going to merge this PR as is because I'm confident that the CURRENT scripts don't support arbitrary bash -- but I'd love to dig into this more, because my new version of the scripts... maybe do work for this case? ???

Reply:
That's not related to Bash. The subprocesses don't get killed properly like you would expect in an init system. This would be true for any subprocess (e.g., if the go binary spawned a subprocess)